dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java

```diff
@@ -206,37 +242,97 @@ protected String makeFieldKey(String schema, String element, String qualifier)
-                ChoiceAuthority ma = (ChoiceAuthority)
-                    pluginService.getNamedPlugin(ChoiceAuthority.class, configurationService.getProperty(key));
-                if (ma == null)
-                {
-                    log.warn("Skipping invalid configuration for "+key+" because named plugin not found: "+configurationService.getProperty(key));
-                    continue;
-                }
-                controller.put(fkey, ma);
-
-                log.debug("Choice Control: For field="+fkey+", Plugin="+ma);
-            }
+            loadChoiceAuthorityConfigurations();
         }
 
         return controller;
     }
 
+	private void loadChoiceAuthorityConfigurations() {
+		// Get all configuration keys starting with a given prefix
+		List<String> propKeys = configurationService.getPropertyKeys(CHOICES_PLUGIN_PREFIX);
+		Iterator<String> keyIterator = propKeys.iterator();
+		while(keyIterator.hasNext())
+		{
+		    String key = keyIterator.next();
+		    String fkey = config2fkey(key.substring(CHOICES_PLUGIN_PREFIX.length()));
+		    if (fkey == null)
+		    {
+		        log.warn("Skipping invalid ChoiceAuthority configuration property: "+key+": does not have schema.element.qualifier");
+		        continue;
+		    }
+
+		    // XXX FIXME maybe add sanity check, call
+		    // MetadataField.findByElement to make sure it's a real field.
+		    String authorityName = configurationService.getProperty(key);
+			ChoiceAuthority ma = (ChoiceAuthority)
+		        pluginService.getNamedPlugin(ChoiceAuthority.class, authorityName);
+		    if (ma == null)
+		    {
+		        log.warn("Skipping invalid configuration for "+key+" because named plugin not found: "+authorityName);
+		        continue;
+		    }
+		    if(!authorities.containsKey(authorityName)) {
+		    	controller.put(fkey, ma);
+		    	authorityNames.add(authorityName);
+		    	authorities.put(authorityName, fkey);
+		    }
+		    else {
+		    	log.warn("Skipping invalid configuration for "+key+" because plugin is alredy in use: "+authorityName +" used by " + authorities.get(authorityName));
+		        continue;		    	
+		    }
+		    
+		    log.debug("Choice Control: For field="+fkey+", Plugin="+ma);
+		}
+		autoRegisterChoiceAuthorityFromInputReader();
+	}
+
+	private void autoRegisterChoiceAuthorityFromInputReader() {
+		try {
+			DCInputsReader dcInputsReader = new DCInputsReader();
+			for (DCInputSet dcinputSet : dcInputsReader.getAllInputs(Integer.MAX_VALUE, 0)) {
+				DCInput[] dcinputs = dcinputSet.getFields();
+				for (DCInput dcinput : dcinputs) {
+					if (StringUtils.isNotBlank(dcinput.getPairsType())
+							|| StringUtils.isNotBlank(dcinput.getVocabulary())) {
+						String authorityName = dcinput.getPairsType();
+						if(StringUtils.isBlank(authorityName)) {
+							authorityName = dcinput.getVocabulary();
+						}
+						if (!StringUtils.equals(dcinput.getInputType(), "qualdrop_value")) {
```

Comment:
This seems like too much submission form config knowledge in the `ChoisesAuthorityService`.

Can't you reuse `DCInputAuthority.getPluginNames` here or move this logic to that class?

Reply:
No, it is not possible use another option to retrieve authority name. Take in mind that the plugin DCInputAuthority and DSpaceControlledVocabulary are selfnamed and we have easier the configuration to autoregister without write configuration into cfg files. Only the submission-forms.xml (a.k.a. input-forms.xml) know how and which authority will be registered.

Reply:
Ok then I think there are two options:
1. Instead of making the `ChoiceAuthorityServiceImpl` directly use a `DCInputsReader`, can't you auto-register (an updated version of) `DCInputAuthority` and `DSpaceControlledVocabulary` instead?
2. If not possible, we should remove `DCInputAuthority` and `DSpaceControlledVocabulary` as they don't have a purpose anymore I think. In that case, we also need to write migration documentation for people currently using those classes.

Reply:
I think that the issue here come from the use of the DSpace plugin framework. We cannot auto-register a ChoiceAuthority in the ChoiceAuthorityServiceImpl because the choice authorities are retrieved via the dspace plugin framework that doesn't support "autoregistration".  The code can be most probably cleanup if we remove the use of the plugin framework to manage ChoiceAuthority in favor of Spring IoC. I have created a separate issue for that: https://jira.duraspace.org/browse/DS-3750 (please note that I haven't flagged it for 7.0 as it is not IMHO a must to have)

Reply:
DCInputAuthority and DSpaceControlledVocabulary can still be used if someone want to expose a value-pairs or a vocabulary as an authorities without bind it to a metadata in the submission-form. It is an edge scenario but we cannot exclude it.  In addition, keeping them in place make this PR smaller (and it is already so big :) )

Reply:
While I'm not really happy with it, I agree that this can serve as a first iteration.